docs/src/main/asciidoc/getting-started-testing.adoc

```diff
@@ -228,6 +228,118 @@ public class StaticContentTest {
 
 For now `@TestHTTPResource` allows you to inject `URI`, `URL` and `String` representations of the URL.
 
+== Testing a specific endpoint
+
+Both RESTassured and `@TestHTTPResource` allow you to specify the endpoint class you are testing rather than hard coding
+a path. This currently supports both JAX-RS endpoints and Servlets. This makes it a lot easier to see exactly which endpoints
+a given test is testing.
+
+For the purposes of these examples I am going to assume we have an endpoint that looks like the following:
+
+[source,java]
+----
+@Path("/hello")
+public class GreetingResource {
+
+    @GET
+    @Produces(MediaType.TEXT_PLAIN)
+    public String hello() {
+        return "hello";
+    }
+}
+----
+
+NOTE: This currently does not support the `@ApplicationPath()` annotation to set the JAX-RS context path. Use the
+`quarkus.resteasy.path` config value instead if you want a custom context path.
+
+=== TestHTTPResource
+
+You can the use the `io.quarkus.test.common.http.TestHTTPEndpoint` annotation to specify the endpoint path, and the path
+will be extracted from the provided endpoint. If you also specify a value for the `TestHTTPResource` endpoint it will
+be appended to the end of the endpoint path.
+
+[source,java]
+----
+package org.acme.getting.started.testing;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URL;
+import java.nio.charset.StandardCharsets;
+
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+import io.quarkus.test.common.http.TestHTTPEndpoint;
+import io.quarkus.test.common.http.TestHTTPResource;
+import io.quarkus.test.junit.QuarkusTest;
+
+@QuarkusTest
+public class StaticContentTest {
+
+    @TestHTTPEndpoint(GreetingResource.class)  // <1>
+    @TestHTTPResource
+    URL url;
+
+    @Test
+    public void testIndexHtml() throws Exception {
+        try (InputStream in = url.openStream()) {
+            String contents = readStream(in);
+            Assertions.assertTrue(contents.equals("hello"));
+        }
+    }
+
+    private static String readStream(InputStream in) throws IOException {
+        byte[] data = new byte[1024];
+        int r;
+        ByteArrayOutputStream out = new ByteArrayOutputStream();
+        while ((r = in.read(data)) > 0) {
+            out.write(data, 0, r);
+        }
+        return new String(out.toByteArray(), StandardCharsets.UTF_8);
+    }
+}
+----
+<1> Because `GreetingResource` is annotated with `@Path("/hello")` the injected URL
```

Comment:
Does it take into account the root path? And the potential `@ApplicationPath`?

Reply:
`@ApplicationPath` is mentioned in the doc to not be supported

Reply:
`quarkus.resteasy.path` is supported

Reply:
OK, sorry for the noise then :).

Reply:
No worries :)

Reply:
I could not really think of a way to handle it in native mode that was not just re-indexing everything. It seemed like a small caveat to an otherwise nice feature though. 

When this is in should we change our quickstarts and guides to use it?

Reply:
Yeah, that would be the best to make it known to users

Reply:
@stuartwdouglas are you planning on doing the follow up for reactive routes, or should I?

Reply:
I am not doing anything more tonight, if you don't do it I will look at it tomorrow. We could also do Websockets, although they support URI templates in their path so it would need to just register the constant part of the template.

Reply:
I am lowered the revs on the engine with PTO approaching, so I may or may not do it today :). In any case I'll let you know.

Reply:
I will just do it tomorrow, its simple and there is no hurry.

Reply:
Yeah I know. The rest of my day is probably going to be pretty light so I'll most likely do it

Reply:
I didn't realize it would be that easy :) https://github.com/quarkusio/quarkus/pull/11092